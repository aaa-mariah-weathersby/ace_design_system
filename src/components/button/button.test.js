import React from 'react';

import PrimaryButton from './primaryButton';
import ButtonBase from '@material-ui/core/Button';

import { shallow, mount } from 'enzyme';


describe('Primary Button', () => {

    // const component = shallow(
    //     <PrimaryButton 
    //         onClick={clickListener}
    //     />
    // )

    test('calls onClick function on click behavior', () => {

        const component = shallow(<PrimaryButton onClick={clickListener}/>)
        const clickListener = jest.fn()

        component.simulate('click')

        const clickListCount = clickListener.mock.calls.length
        expect(clickListCount).toBe(1);

    });

})

