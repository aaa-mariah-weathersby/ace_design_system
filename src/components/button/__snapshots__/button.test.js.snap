// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component 1`] = `
<button
  className="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  />
  <span
    className="MuiTouchRipple-root-28"
  />
</button>
`;

exports[`Primary Button calls onClick function 1`] = `
<button
  className="MuiButtonBase-root-25 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  />
  <span
    className="MuiTouchRipple-root-28"
  />
</button>
`;
